lib/src/element_type.dart

```diff
@@ -26,11 +26,15 @@ abstract class ElementType extends Privacy with CommentReferable, Nameable {
 
   factory ElementType.from(
       DartType f, Library library, PackageGraph packageGraph,
-      [ElementType returnedFrom]) {
+      {ElementType returnedFrom}) {
     if (f.element == null ||
         f.element.kind == ElementKind.DYNAMIC ||
         f.element.kind == ElementKind.NEVER) {
       if (f is FunctionType) {
+        if (f.aliasElement != null) {
+          return AliasedFunctionTypeElementType(
```

Comment:
Null safety suggestion: make `aliasElement` and `aliasArguments` required parameters of `AliasedFunctionTypeElementType`. It is nice to use the type system to prove that some data is available. Here, we have already proved that `aliasElement` is available. But then we drop this proof, and force `AliasedFunctionTypeElementType` to implicitly require that `aliasElement` and `aliasArguments` are not `null`.

Reply:
Wow, this is a very thorough explanation, thank you for it.

I appreciate the intention to keep code consistent, and was merely sharing my view on writing more null safe code. But DartDoc is not null safe, so it might be premature to do anyway :-)